src/gt4py/next/otf/binding/nanobind.py

```diff
@@ -187,7 +187,47 @@ def make_argument(name: str, type_: ts.TypeSpec) -> str | BufferSID | CompositeS
     elif isinstance(type_, ts.ScalarType):
         return name
     else:
-        raise ValueError(f"Type '{type_}' is not supported in pybind11 interfaces.")
+        raise ValueError(f"Type '{type_}' is not supported in nanobind interfaces.")
+
+
+nanobind_sid_conv = """\
+#include <nanobind/nanobind.h>
+#include <nanobind/ndarray.h>
+#include <gridtools/sid/synthetic.hpp>
+#include <gridtools/sid/simple_ptr_holder.hpp>
+#include <gridtools/sid/unknown_kind.hpp>
+#include <gridtools/common/integral_constant.hpp>
+#include <gridtools/common/array.hpp>
+#include <algorithm>
+
+
+namespace gridtools {
+    namespace nanobind_sid_adapter_impl_ {
+        template <size_t, class>
+        struct kind {};
+
+        template <std::size_t UnitStrideDim = std::size_t(-1), class T, std::size_t... Sizes, class... Args>
+        auto as_sid(nanobind::ndarray<T, nanobind::shape<Sizes...>, Args...> ndarray) {
+            using sid::property;
+            const auto ptr = ndarray.data();
+            constexpr auto ndim = sizeof...(Sizes);
+            array<size_t, ndim> shape;
+            array<size_t, ndim> strides;
+            std::copy_n(ndarray.shape_ptr(), ndim, shape.begin());
+            std::copy_n(ndarray.stride_ptr(), ndim, strides.begin());
+
+            return sid::synthetic()
+                .template set<property::origin>(sid::host_device::simple_ptr_holder<T *>{ptr})
+                .template set<property::strides>(strides)
+                .template set<property::strides_kind, kind<ndim, void>>()
```

Comment:
```suggestion
                .template set<property::strides_kind, sid::unknown_kind>()
```
or make kind part of the template parameters and default it to `sid::unknown_kind` there.
Your choice would mean all arrays of same dimensionality have the same strides.

Reply:
some failing tests are related to this.

Reply:
Yeah, it works with unknown kind.